src/Database/Type/TimeType.php

```diff
@@ -16,67 +16,90 @@
  */
 namespace Cake\Database\Type;
 
-use Cake\Chronos\Chronos;
-use Cake\I18n\DateTime;
-use DateTime as NativeDateTime;
-use DateTimeImmutable;
+use Cake\Chronos\ChronosTime;
+use Cake\Database\Driver;
+use Cake\I18n\Time;
 use DateTimeInterface;
-use Exception;
-use InvalidArgumentException;
 
 /**
  * Time type converter.
  *
  * Use to convert time instances to strings & back.
  */
-class TimeType extends DateTimeType
+class TimeType extends BaseType implements BatchCastingInterface
 {
     /**
-     * @inheritDoc
+     * The PHP Time format used when converting to string.
+     *
+     * @var string
      */
     protected string $_format = 'H:i:s';
 
     /**
-     * {@inheritDoc}
+     * The ICU Time formats allowed by `marshal()`.
      *
      * @var array<string>
      */
     protected array $_marshalFormats = [
-        'H:i:s',
-        'H:i',
+        'HH:mm:ss',
+        'HH:mm',
     ];
 
+    /**
+     * Whether `marshal()` should use locale-aware parser with `_localeMarshalFormat`.
+     *
+     * @var bool
+     */
+    protected bool $_useLocaleMarshal = false;
+
+    /**
+     * The locale-aware format `marshal()` uses when `_useLocaleParser` is true.
+     *
+     * See `Cake\I18n\Time::parseTime()` for accepted formats.
+     *
+     * @var string|int|null
+     */
+    protected string|int|null $_localeMarshalFormat = null;
+
+    /**
+     * The classname to use when creating objects.
+     *
+     * @var class-string<\Cake\I18n\Time>
+     */
+    protected string $_className;
+
+    /**
+     * @inheritDoc
+     */
+    public function __construct(?string $name = null)
+    {
+        parent::__construct($name);
+
+        $this->_className = Time::class;
```

Comment:
In e.g. the DateTimeType we check if the Chronos class is actually present but since there is no PHP native Time class we can't use a fallback there. But do we want to make this class adjustable by the user?

Reply:
I'm not really in favor of the php DateTime support as it's always been untested.

Reply:
Totally aggree, so should I instead just use the class directly instead of a property here? Looks a bit cleaner to me to be honest.

Reply:
This is for @markstory since that behavior existed before I joined. I don't know if it's important for something.

Reply:
I'm ok with throwing an error if neither I18n\Time or ChronosTime can be found. There are no reasonable fallbacks in PHP that we can use like there are for DateTime. It's arguable that we should also throw errors for DateType as the PHP fallback isn't really the same either.

Reply:
The reason this exists is the split packages. You could have Database without I18n or Chronos still I think.

Reply:
> I'm ok with throwing an error if neither I18n\Time or ChronosTime can be found. There are no reasonable fallbacks in PHP that we can use like there are for DateTime. It's arguable that we should also throw errors for DateType as the PHP fallback isn't really the same either.

But then out 9f the box wouldn't anyone using the standalone database/orm package without the I18n package get an error, since TypeFactory maps time columns to TimeType? So at least should check and fallback to using StringType if I18n\Type or ChronosTime is not available.

Reply:
> But then out of the box wouldn't anyone using the standalone database/orm package without the I18n package get an error, since TypeFactory maps time columns to TimeType? 

We could fallback to chronos objects, or make chronos a hard requirement for the database package and remove all the PHP fallbacks 🤷

Reply:
> ..or make chronos a hard requirement for the database package

I guess that's a better alternative then having errors for date/time columns.

Reply:
I am not sure how the sub-packaged composer.json needs to be adjusted to do that 😅